src/compiler/typescript_to_rust/azle_generate/src/ts_ast/azle_type/azle_type_ref/to_data_type.rs

```diff
@@ -35,8 +35,29 @@ impl AzleTypeRef<'_> {
     }
 
     fn to_type_ref(&self) -> TypeRef {
+        let type_arguments = if let Some(type_params) = &self.ts_type_ref.type_params {
+            type_params
+                .params
+                .iter()
+                .map(|param| {
+                    // TODO is this the right source map to pass in?
```

Comment:
Yes, the source map is good. The source map is tied to a program so the same one should be used for a type_ref as for its type params.